### DISCUSSION TITLE
Staticman in jekyll leads to 405 “not allowed”

### DISCUSSION THREAD

**Discussion by cmeissner** [community/community#23368] [Other Feature Feedback, Questions, & Ideas]
<p>I tried to configure ‘staticman’ comments for my github pages. I use Beautiful Jekyll for my site.</p>
<p>I followed the documentation of <a href="https://staticman.net/docs/getting-started" rel="noopener nofollow ugc">staticman getting-started</a>.</p>
<ol>
<li>created a github application and install it in my <a href="https://github.com/codeaffen/codeaffen.org" rel="noopener nofollow ugc">repo</a>
</li>
<li>deploy staticman app on heroku</li>
<li>configure <code>_config.yml</code>
</li>
<li>configure <code>staticman.yml</code>
</li>
</ol>
<p>Unfortunately I get only a HTTP 405 (not allowd) when I try to submit a comment. There does not arrive any request at my staticman installation. So I would guess this is a problem at github pages side.</p>

**Comment by yoannchaudet** [comment]
<p>Hello!</p>
<p>Any chances you can share with us the exact HTTP request that is failing?</p>
<p>GitHub Pages does not support verbs other than GET for instance. I suspect on your repo, staticman is trying to make non-GET requests against Pages instead of your Heroku app for some reasons.</p>

**Comment by cmeissner** [comment]
<p>If I try to to place a comment <a href="https://staging.codeaffen.org/2021-07-16-oh-my-zsh-walkthrough/" rel="noopener nofollow ugc">here</a> having Developer Console open I saw that the form seem to sends the post request to itself.<br>
I am not so good in jquery and javascript to understand what should happen instead. What I can say is that the jekyll rendering will work fine the staticman.js is loaded but my endpoint is never called from the form.</p>

**Comment by yoannchaudet** [comment]
<p>I have never used staticman but I think you are simply missing an <code>action</code> attribute on your <code>form</code> tag:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/d/3da2d14485e57087474b350591e5e4ae380217a6.png" data-download-href="/uploads/short-url/8NfXy54hNNn7U82gVOy5bqeLSCO.png?dl=1" title="image"><img src="https://user-images.githubusercontent.com/14911070/181076579-09429fea-9277-4183-a2f5-bde175740fd5.png" alt="image" data-base62-sha1="8NfXy54hNNn7U82gVOy5bqeLSCO" width="690" height="333" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/d/3da2d14485e57087474b350591e5e4ae380217a6_2_690x333.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/d/3da2d14485e57087474b350591e5e4ae380217a6_2_1035x499.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/d/3da2d14485e57087474b350591e5e4ae380217a6_2_1380x666.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/d/3da2d14485e57087474b350591e5e4ae380217a6_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">3688×1782 562 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>This should point to your staticman service (the one you deployed if following the documentation as a Heroku app). This is step 4 of the <a href="https://staticman.net/docs/getting-started">guide</a>.</p>

**Comment by cmeissner** [comment]
<p>Great advice but the jekyll theme (beautifuljekyll) do the trick by using jquery.js and staticman.js. So I would guess here the .submit should do the trick. No idea why this is not done.</p>

**Comment by cmeissner** [comment, marked as answer]
<p>I found an open <a href="https://github.com/daattali/beautiful-jekyll/issues/766" rel="noopener nofollow ugc">bug on beautifuljekyll</a> which describe the whole behavior. I’ll have an eye on it.</p>
